controllers/projectController.js

```diff
@@ -31,24 +35,55 @@ const setProject = asyncHandler(async (req, res) => {
       your_impact: req.body.your_impact,
       brand_color: req.body.brand_color,
       main_image: req.files.main_image[0].filename,
-      main_image_url: req.files.main_image[0].path,
+      main_image_url: req.files.main_image[0].path.replace('public/', ''),
       background_image: req.files.background_image[0].filename,
-      background_image_url: req.files.background_image[0].path,
+      background_image_url: req.files.background_image[0].path.replace('public/', ''),
     })
     res.status(200).json(project)
   } catch (error) {
+    res.status(400)
     throw new Error(error)
   }
 })
 
 // PUT  update project  /api/project/:id
 const updateProject = asyncHandler(async (req, res) => {
   const project = await Project.findById(req.params.id)
+
   if(!project) {
     res.status(400)
     throw new Error("Project not found")
   }
-  const updatedProject = await Project.findByIdAndUpdate(req.params.id, req.body, {
+
+  let mainImage = project.main_image
+  let mainImageUrl = project.main_image_url
+  let backgroundImage = project.background_image
+  let backgroundImageUrl = project.background_image_url
+
+  if(req?.files?.main_image) {
+    mainImage = req.files.main_image[0].filename
+    mainImageUrl = req.files.main_image[0].path.replace('public/', '')
+    await unlinkAsync(`public/${project.main_image_url}`)
+  }
+  if(req?.files?.background_image) {
+    backgroundImage = req.files.background_image[0].filename
+    backgroundImageUrl = req.files.background_image[0].path.replace('public/', '')
+    await unlinkAsync(`public/${project.background_image_url}`)
```

Comment:
Let's wrap deleting file in try/catch block, if error occurs -> respond with 500 code and some message